test/integration/roundtrip_test.go

```diff
@@ -0,0 +1,291 @@
+/*
```

Comment:
Test heavily wip.

Please define round-trip so we're on the same page. I'm going to put a broker in and make sure I get the same broker out. And actually have any checking at all in the test.